src/app/account/new-order/new-order.component.ts

```diff
@@ -619,4 +623,9 @@ export class NewOrderComponent implements OnInit, OnDestroy {
     };
     return type.name;
   }
+
+  public billingRequired(): boolean {
+    return !this.config.config?.noBillingForFreeOrder ||
+      !this.products.every(product => product.pricing?.pricing_type === "FREE");
```

Comment:
if pricing_type always explicitly set, or can it be empty?
If empty, I would consider empty to be same as "FREE".